RecoParticleFlow/PFRecHitProducer/plugins/alpaka/CalorimeterDefinitions.h

```diff
@@ -9,12 +9,15 @@
 #include "DataFormats/HcalRecHit/interface/HBHERecHit.h"
 #include "DataFormats/ParticleFlowReco/interface/CaloRecHitHostCollection.h"
 #include "DataFormats/ParticleFlowReco/interface/alpaka/CaloRecHitDeviceCollection.h"
-#include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitParamsRecord.h"
 #include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitTopologyHostCollection.h"
-#include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitTopologyRecord.h"
 #include "RecoParticleFlow/PFRecHitProducer/interface/alpaka/PFRecHitParamsDeviceCollection.h"
 #include "RecoParticleFlow/PFRecHitProducer/interface/alpaka/PFRecHitTopologyDeviceCollection.h"
 
+class PFRecHitHCALTopologyRecord;
+class PFRecHitECALTopologyRecord;
+class PFRecHitHCALParamsRecord;
+class EcalPFRecHitThresholdsRcd;
```

Comment:
I'm wondering why these forward declarations are needed at all. The kernel code should not need these types declared. The ED/ES module side would need the definitions of these classes in any case (except for the case for only declaring `ESGetToken`s).

Reply:
Not sure I understood your question. These classes are used a few lines down (L33-40, L112-119) to define type aliases inside `HCAL` and `ECAL` structures, and these aliases are used by a template class `PFRecHitSoAProducer`, which is parametrized by calorimeter type.

Reply:
Ah right. I find it unfortunate that the EDProducer-side and the kernel-side information + functions are coupled like this, but ok.